presto-hive/src/test/java/io/prestosql/plugin/hive/TestHiveIntegrationSmokeTest.java

```diff
@@ -6336,6 +6336,64 @@ public void testAnalyzeUnpartitionedTable()
         assertUpdate("DROP TABLE " + tableName);
     }
 
+    @Test
+    public void testAnalyzeTableTimestampWithPrecision()
+    {
+        String catalog = getSession().getCatalog().get();
+        Session nanosecondsTimestamp = Session.builder(withTimestampPrecision(getSession(), HiveTimestampPrecision.NANOSECONDS))
+                // Disable column statistics collection when creating the table
+                .setCatalogSessionProperty(catalog, "collect_column_statistics_on_write", "false")
+                .build();
+        Session microsecondsTimestamp = withTimestampPrecision(getSession(), HiveTimestampPrecision.MICROSECONDS);
+        Session millisecondsTimestamp = getSession();
+
+        String tableName = "test_analyze_timestamp_with_precision";
+
+        try {
+            assertUpdate(nanosecondsTimestamp,
+                    "CREATE TABLE " + tableName + "(c_timestamp) AS VALUES " +
+                            "TIMESTAMP '1988-04-08 02:03:04.111', " +
+                            "TIMESTAMP '1988-04-08 02:03:04.115', " +
+                            "TIMESTAMP '1988-04-08 02:03:04.115', " +
+                            "TIMESTAMP '1988-04-08 02:03:04.119', " +
+                            "TIMESTAMP '1988-04-08 02:03:04.111111', " +
+                            "TIMESTAMP '1988-04-08 02:03:04.111115', " +
+                            "TIMESTAMP '1988-04-08 02:03:04.111115', " +
+                            "TIMESTAMP '1988-04-08 02:03:04.111999', " +
+                            "TIMESTAMP '1988-04-08 02:03:04.111111111', " +
+                            "TIMESTAMP '1988-04-08 02:03:04.111111115', " +
+                            "TIMESTAMP '1988-04-08 02:03:04.111111115', " +
+                            "TIMESTAMP '1988-04-08 02:03:04.111111999' ",
+                    12);
+
+            assertQuery("SHOW STATS FOR " + tableName,
+                    "SELECT * FROM VALUES " +
+                            "('c_timestamp', null, null, null, null, null, null), " +
+                            "(null, null, null, null, 12.0, null, null)");
+
+            assertUpdate(nanosecondsTimestamp, "ANALYZE " + tableName, 12);
+            assertQuery("SHOW STATS FOR " + tableName,
+                    "SELECT * FROM VALUES " +
+                            "('c_timestamp', null, 9.0, 0.0, null, null, null), " +
+                            "(null, null, null, null, 12.0, null, null)");
+
+            assertUpdate(microsecondsTimestamp, "ANALYZE " + tableName, 12);
```

Comment:
It would be cleaner to drop the stats before running ANALYZE the second time, or create a new table

Reply:
I added dropping